src/nvim/screen.c

```diff
@@ -6284,6 +6286,9 @@ void grid_alloc(ScreenGrid *grid, int rows, int columns, bool copy, bool valid)
   new.Rows = rows;
   new.Columns = columns;
 
+  new.comp_height = rows;
```

Comment:
Makes sense, the reason I added these lines here is because tests would fail without it, and I didn't really know where in `ui_compositor.c` they should have gone instead.

Reply:
I ended up moving these lines to `screenalloc()` since I couldn't find a suitable place to initialize `comp_height/width` in ui_compositor.c, and tests fail without them.

Reply:
sure, default_grid is a bit special anyway (and should "eventually" be gotten rid of).